src/Microsoft.OData.Core/JsonLight/ODataJsonLightReader.cs

```diff
@@ -2381,7 +2389,8 @@ private abstract class JsonLightResourceBaseScope : Scope, IODataJsonLightReader
                 IEdmStructuredType expectedResourceType,
                 PropertyAndAnnotationCollector propertyAndAnnotationCollector,
                 SelectedPropertiesNode selectedProperties,
-                ODataUri odataUri)
+                ODataUri odataUri,
+                bool resourceNullable = true)

```

Comment:
> = true [](start = 37, length = 7)

Similar to Biao's comment, I'm not generally a fan of optional params if we're not trying to support backward compatibility and it's easy enough to pass the default.